### ISSUE TITLE
Positive Definitness

### ISSUE THREAD

**Issue by cmmenzen** [bonevbs/HssMatrices.jl#18]
Hello Boris,

Does the approximation of a kernel matrix with hss guarantee to preserve positive definiteness? If yes, how does it achieve that?

Thank you!

**Comment by mvsoom** [comment]
OK, thank you very much for your answer.

Since the block diagonal is conserved, I assume the rank of the matrix is expected to be conserved too?

**Comment by bonevbs** [comment]
No, imagine you have a matrix with all entries equal to one. If you set all the off-diagonal blocks to 0, your rank will have increased from one to something higher.
